include/SFML/Graphics/Rect.inl

```diff
@@ -25,116 +25,146 @@
 }
 
 
 ////////////////////////////////////////////////////////////
 template <typename T>
-constexpr std::optional<Rect<T>> Rect<T>::findIntersection(const Rect<T>& rectangle) const
+constexpr Vector2<T> Rect<T>::getStart() const
 {
-    // Not using 'std::min' and 'std::max' to avoid depending on '<algorithm>'
-    const auto min = [](T a, T b) { return (a < b) ? a : b; };
-    const auto max = [](T a, T b) { return (a < b) ? b : a; };
+    return Vector2<T>(getLeft(), getTop());
+}
 
-    // Rectangles with negative dimensions are allowed, so we must handle them correctly
 
-    // Compute the min and max of the first rectangle on both axes
-    const T r1MinX = min(left, static_cast<T>(left + width));
-    const T r1MaxX = max(left, static_cast<T>(left + width));
-    const T r1MinY = min(top, static_cast<T>(top + height));
-    const T r1MaxY = max(top, static_cast<T>(top + height));
+////////////////////////////////////////////////////////////
+template <typename T>
+constexpr Vector2<T> Rect<T>::getEnd() const
+{
+    return Vector2<T>(getRight(), getBottom());
+}
 
-    // Compute the min and max of the second rectangle on both axes
-    const T r2MinX = min(rectangle.left, static_cast<T>(rectangle.left + rectangle.width));
-    const T r2MaxX = max(rectangle.left, static_cast<T>(rectangle.left + rectangle.width));
-    const T r2MinY = min(rectangle.top, static_cast<T>(rectangle.top + rectangle.height));
-    const T r2MaxY = max(rectangle.top, static_cast<T>(rectangle.top + rectangle.height));
 
-    // Compute the intersection boundaries
-    const T interLeft   = max(r1MinX, r2MinX);
-    const T interTop    = max(r1MinY, r2MinY);
-    const T interRight  = min(r1MaxX, r2MaxX);
-    const T interBottom = min(r1MaxY, r2MaxY);
+////////////////////////////////////////////////////////////
+template <typename T>
+constexpr Vector2<T> Rect<T>::getCenter() const
+{
+    return position + size / static_cast<T>(2);
+}
 
-    // If the intersection is valid (positive non zero area), then there is an intersection
-    if ((interLeft < interRight) && (interTop < interBottom))
-    {
-        return Rect<T>({interLeft, interTop}, {interRight - interLeft, interBottom - interTop});
-    }
-    else
-    {
-        return std::nullopt;
-    }
+
+////////////////////////////////////////////////////////////
+template <typename T>
+constexpr Vector2<T> Rect<T>::getAbsoluteSize() const
+{
+    return Vector2<T>(getRight() - getLeft(), getBottom() - getTop());
+}
+
+
+////////////////////////////////////////////////////////////
+template <typename T>
+constexpr Rect<T> Rect<T>::getAbsoluteRect() const
+{
+    return Rect<T>(getStart(), getAbsoluteSize());
+}
+
+
+////////////////////////////////////////////////////////////
+template <typename T>
+constexpr bool Rect<T>::contains(const Vector2<T>& point) const
+{
+    return getLeft() <= point.x && point.x <= getRight() && getTop() < point.y && point.y < getBottom();
```

Comment:
See, if we *un-delegate* the negative size safeness from the 4 methods, it will have to be moved to every method that uses them! We are going to have to create an entire Rect instance, initialized with an absoluteRect(), even if we need to determine *one* edge (or write the questionmark expression ourselves), which is just so inconvinient! The entire architecture shatters!